docs/consensus.qbk

```diff
@@ -0,0 +1,651 @@
+
+
+[h6 Establish]
+
+The establish phase is the active period of consensus in which the node
+exchanges proposals with peers in an attempt to reach agreement on the consensus
+transactions and effective close time.  
+
+A call to =startRound= would forcibly begin the next consensus round, skipping
+completion of the current round.  This is not expected during normal operation.
+Calls to =peerProposal= or =gotTxSet= that reflect new positions will generate
+disputed transactions for any new disagreements and will update the peer's vote
+for all disputed transactions.
+
+A call to =timerEntry= first checks that the node is working from the correct
+prior ledger. If not, the node  will update the mode and request the correct
+ledger.  Otherwise, the node updates the node's position and considers whether
+to switch to the =Accepted= phase and declare consensus reached.  However, at
+least =LEDGER_MIN_CONSENSUS= time must have elapsed before doing either.  This
+allows peer's an opportunity to take an initial position and share it.
+
+['Update Position]
+
+In order to achieve consensus, the node is looking for a transaction set that is
+supported by a super-majority of peers.  The node works towards this set by
+adding or removing disputed transactions from its position based on an
+increasing threshold for inclusion.
+
+[$images/consensus/threshold.png  [width 50%] [height 50%]]
+
+By starting with a lower threshold, a node initially allows a wide set of
+transactions into its position. If the establish round continues and the node is
+"stuck", a higher threshold can focus on accepting transactions with the most
+support.  The constants that define the thresholds and durations at which the
+thresholds change are given by `AV_XXX_CONSENSUS_PCT` and
+`AV_XXX_CONSENSUS_TIME` respectively, where =XXX= is =INIT=,=MID=,=LATE= and
+=STUCK=.  The effective close time position is updated using the same
+thresholds.
+
+Given the example disputes above and an initial threshold of 50%, our node would
+retain its position since transaction 1 was not in dispute and transactions 2
+and 3 have 75% support.  Since its position did not change, it would not need to
+send a new proposal to peers.  Peer C would not change either. Peer A would add
+transaction 3 to its position and Peer B would remove transaction 4 from its
+position; both would then send an updated position. 
+
+Conversely, if the diagram reflected a later call to =timerEntry= that occurs in
+the stuck region with a threshold of say 95%, our node would remove transactions
+2 and 3 from its candidate set and send an updated position.  Likewise, all the
+other peers would end up with only transaction 1 in their position.
+
+Lastly, if our node were not in the proposing mode, it would not include its own
+vote and just take the majority (>50%) position of its peers. In this example,
+our node would maintain its position of transactions 1, 2 and 3.
+
+['Checking Consensus]
```

Comment:
I like these subsections for the actions / transitions, and think they could do well under the other states.